app/lib/import_export.rb

```diff
@@ -11,497 +12,186 @@ module ImportExport # rubocop:todo Metrics/ModuleLength
-                                                          extended_isolation end_of_monitoring closed_at monitoring_reason expected_purge_ts]),
-              rct_node(:patients, 'Reporting Info', %i[responder_id head_of_household pause_notifications last_assessment_reminder_sent])
-            ]
-          }
-        ]
-      }
-    ]
-  }.freeze
-
-  ASSESSMENTS_EXPORT_OPTIONS = {
-    label: 'Reports',
-    nodes: [rct_node(:assessments, 'Reports', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id symptomatic
-                                                 who_reported created_at updated_at symptoms])]
-  }.freeze
-
-  LABORATORIES_EXPORT_OPTIONS = {
-    label: 'Lab Results',
-    nodes: [rct_node(:laboratories, 'Lab Results', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id lab_type
-                                                      specimen_collection report result created_at updated_at])]
-  }.freeze
-
-  CLOSE_CONTACTS_EXPORT_OPTIONS = {
-    label: 'Close Contacts',
-    nodes: [rct_node(:close_contacts, 'Close Contacts', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id first_name
-                                                           last_name primary_telephone email contact_attempts last_date_of_exposure assigned_user notes
-                                                           enrolled_id created_at updated_at])]
-  }.freeze
-
-  TRANSFERS_EXPORT_OPTIONS = {
-    label: 'Transfers',
-    nodes: [rct_node(:transfers, 'Transfers', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id who from_jurisdiction
-                                                 to_jurisdiction created_at updated_at])]
-  }.freeze
-
-  HISTORIES_EXPORT_OPTIONS = {
-    label: 'History',
-    nodes: [rct_node(:histories, 'History', %i[patient_id user_defined_id_statelocal user_defined_id_cdc user_defined_id_nndss id created_by history_type
-                                               comment created_at updated_at])]
-  }.freeze
-
-  CUSTOM_EXPORT_OPTIONS = {
-    patients: PATIENTS_EXPORT_OPTIONS,
-    assessments: ASSESSMENTS_EXPORT_OPTIONS,
-    laboratories: LABORATORIES_EXPORT_OPTIONS,
-    close_contacts: CLOSE_CONTACTS_EXPORT_OPTIONS,
-    transfers: TRANSFERS_EXPORT_OPTIONS,
-    histories: HISTORIES_EXPORT_OPTIONS
-  }.freeze
+  # Finds the symptoms needed for the reports columns
+  def update_assessment_symptom_data(data)
+    # Don't update if assessment symptom data isn't needed
+    return unless data.dig(:assessments, :checked)&.include?(:symptoms)
+
+    # Using Symptom.all for finding distinct symptom names is actually faster here because it can use symptoms_index_chain_1
+    symptom_names = Symptom.all.distinct.pluck(:name).sort
+    symptom_labels = symptom_names.map { |name| Symptom.find_by(name: name).label }
```

Comment:
Yes, it's much faster to execute these two lines (even though the second one loops through n queries) than to do a `pluck(:name, :label)` or a `pluck(:label)` because `name` is indexed and `label` is not

Reply:
Should we just add an index for this rather than taking this kind of roundabout approach then?

Reply:
hmm, I think that's a question for @holmesie 

not sure if it's worth adding an index on symptoms (since there are so many of them) and the index would only be used for this one use case and it wouldn't really speed it up much more than just querying by `name`.

Reply:
this code is no longer relevant as the implementation of this method has changed: https://github.com/SaraAlert/SaraAlert/pull/717/commits/ab0472956fb88101b83558e527cd1558fd81a1c0#diff-c34521fcd0fdc2034d456886aae8fd35299fe908ab88d8e49c60d0b4d39ab26bR193